#49 Simple kubecfg.yaml endpoint security.

joaovrodrigues opened the pull request:
Added simple verification to the kubecfg.yaml endpoint, should show the 404 page when user not logged in and by doing so preventing partial credentials from being exposed publicly.

negz commented on extractor/oidc.go:
```diff
@@ -97,20 +99,22 @@ func (o *oidcExtractor) Process(ctx context.Context, cfg *oauth2.Config, code st
 		return nil, errors.Wrap(err, "cannot verify ID token")
 	}
 
+	fmt.Println(idt)
```
This seems like it might be leftover from debugging.

negz commented on extractor/oidc.go:
```diff
@@ -97,20 +99,22 @@ func (o *oidcExtractor) Process(ctx context.Context, cfg *oauth2.Config, code st
 		return nil, ErrMissingIDToken
 	}
-	o.log.Debug("token", zap.String("id", id), zap.Any("token", token))
-
```
Why were these debug log lines removed?

negz commented on extractor/oidc.go:
```diff
@@ -119,5 +123,7 @@ func (o *oidcExtractor) Process(ctx context.Context, cfg *oauth2.Config, code st
 		return nil, errors.New("Invalid email domain, expecting " + o.emailDomain)
 	}
 
+	o.log.Error("UserInfo", zap.Any("userinfo", params))
```
This does not seem to be an error condition, but it is being logged as an error. Should it be a debug log line instead?

negz commented on kuberos.go:
```diff
@@ -3,23 +3,30 @@ package kuberos
+	"flag"
 	"fmt"
 	"io/ioutil"
+	"k8s.io/client-go/kubernetes"
```
Imports should be sorted such that stdlib imports are separated by a blank line from third party imports.

negz commented on kuberos.go:
```diff
@@ -224,9 +231,15 @@ func (h *Handlers) Login(w http.ResponseWriter, r *http.Request) {
 
 	u := c.AuthCodeURL(h.state(r), h.oo...)
 	h.log.Debug("redirect", zap.String("url", u))
+
```
This seems like a spurious whitespace change.

negz commented on kuberos.go:
```diff
@@ -224,9 +231,15 @@ func (h *Handlers) Login(w http.ResponseWriter, r *http.Request) {
+//Will create user admin RBAC
+func createUserRole() {
+
+}
```
This doesn't seem to do anything.

negz commented on kuberos.go:
```diff
@@ -317,13 +335,16 @@ func Template(cfg *api.Config) http.HandlerFunc {
 			http.Error(w, errors.Wrap(err, "cannot parse URL parameter").Error(), http.StatusBadRequest)
 			return
+		} else if len(p.Username) == 0 {
+			http.Error(w, http.StatusText(http.StatusNotFound), http.StatusNotFound)
```
I'd prefer to return HTTP 401 Unauthorized in this case.

negz reviewed (changes requested):
Thanks for tackling this!

I've taken a first pass review. It seems like this PR needs a bit of cleanup before it can be merged; there's a few `fmt` statements that I assume were intended for debugging, as well as a bunch of spurious whitespace changes and permission changes on the shell scripts. Please clean these things up and I'll take another look.

joaovrodrigues commented:
So I accidentally added a commit from experimenting/exploring with Kubernetes, go-client and OIDC where I added/removed a bunch of log lines and code as I'm making my way through this tech stack.  Removed that commit as it's not part of the scope of this PR. 

Permission changes are a result from me moving away from macOS and tweaking with WSL for the scripts to work, also should not be in this commit.

I personally prefer to to have 404 in pages that shouldn't be accessible from the outside, and believe it's industry standard as well, but 401 is also a valid status code, changed it to that.
